### PR TITLE
[FIX JENKINS-50861] update core and Pipeline scm step dependencies

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-50861](https://issues.jenkins-ci.org/browse/JENKINS-50861)

* Update Pipeline SCM step to 2.3
* Update Pipeline to 2.5 for test
* Update core to 2.60
* Update Java to 8
* Update SCM API  to 2.2.6
* Update credentials to 2.1.16
* fix inherited dependencies
* propagate new exceptions

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -256,9 +262,18 @@ THE SOFTWARE.
...
+      <version>${workflow-aggregator.version}</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jenkins-ci.plugins.workflow</groupId>
+          <artifactId>workflow-step-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.jenkins-ci.plugins</groupId>
+          <artifactId>script-security</artifactId>
```
Comment (dwnusbaum) [inline pom.xml]: One of the CI tests is failing with `java.lang.NoClassDefFoundError: org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval`, so maybe these shouldn't be excluded?

Comment (kuisathaverat) [reply pom.xml]: seems like, I have had some problems with maven-enforcer-plugin to correct the dependencies.

**Comment by oleg-nenashev** [comment]
I would propose to merge #216 which fixes the build at least. Then @kuisathaverat can enjoy playing with Pipeline deps :)

File: src/main/java/jenkins/scm/impl/subversion/SubversionSCMSource.java [hunk 2]
```diff
@@ -231,8 +231,7 @@ public synchronized String getUuid() {
     protected void retrieve(@CheckForNull SCMSourceCriteria criteria,
                             @NonNull final SCMHeadObserver observer,
                             @CheckForNull SCMHeadEvent<?> event,
-                            @NonNull TaskListener listener)
-            throws IOException {
+                            @NonNull TaskListener listener) throws IOException, InterruptedException {
```
Comment (dwnusbaum) [inline src/main/java/jenkins/scm/impl/subversion/SubversionSCMSource.java]: Do we still need to add `InterruptedException` or was that just required for a dependency update?

File: pom.xml [hunk 3]
```diff
@@ -138,9 +138,10 @@ THE SOFTWARE.
     <jenkins.version>2.60.3</jenkins.version>
     <java.level>8</java.level>
     <no-test-jar>false</no-test-jar>
-    <workflow.version>1.14.2</workflow.version>
+    <workflow-scm-step.version>2.6</workflow-scm-step.version>
```
[the remark this answers is not in the document]
Comment (kuisathaverat) [reply pom.xml]: I think that always should be a difference between the workflow-scm-step version used and the  workflow-aggregator version.

File: pom.xml [hunk 4]
```diff
@@ -256,9 +263,18 @@ THE SOFTWARE.
     <dependency>
       <groupId>org.jenkins-ci.plugins.workflow</groupId>
       <artifactId>workflow-aggregator</artifactId>
-      <version>${workflow.version}</version>
-      <classifier>tests</classifier>
+      <version>${workflow-aggregator.version}</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
```
[the remark this answers is not in the document]
Comment (dwnusbaum) [reply pom.xml]: These exclusions can be removed as well I think.

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dwnusbaum) [reply pom.xml]: Ah, ok

File: pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kuisathaverat) [reply pom.xml]: the problem here, it is if we include the dependency we probably have to maintain two new variables for the workflow-* plugins and each time we upgrade on we would have to fight again to fix the dependencies

[the remark this answers is not in the document]
Comment (dwnusbaum) [reply pom.xml]: Well either way, this is a test scope dependency, so it should be fine for now.

**Comment by kuisathaverat** [comment]
@reviewbybees done

**Pull request merged**